plugins/geotiff/main.py

```diff
@@ -0,0 +1,97 @@
+import json
+import os
+
+import rasterio
+from pyproj import Proj
+
+from vistas.core.gis.extent import Extent
+from vistas.core.plugins.data import RasterDataPlugin, VariableStats
+
+
+class GeoTIFF(RasterDataPlugin):
+
+    id = 'geotiff'
+    name = 'GeoTIFF Data Plugin'
+    description = 'Reads GeoTIFF (.tif) files'
+    author = 'Conservation Biology Institute'
+    version = '1.0'
+    extensions = [('tif', 'GeoTIFF')]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._name = None
+        self._shape = None
+        self._extent = None
+        self._resolution = None
+        self._nodata = None
+        self._count = None
+
+    def load_data(self):
+        self._name = self.path.split(os.sep)[-1].split('.tif')[0]
+        with rasterio.open(self.path) as src:
+            projection = Proj(src.crs)
+            self._shape = src.shape
+            self._resolution = src.res[0]
+            xmin, ymin = src.xy(src.height, 0, 'ul')
+            xmax, ymax = src.xy(0, src.width, 'ul')
+            self._extent = Extent(xmin, ymin, xmax, ymax, projection)
+            self._nodata = src.nodata
+            self._count = src.count
+
+    @property
+    def data_name(self):
```

Comment:
I think from an API point of view, I think as long as we're clear on what methods need to be overridden, then we could go either way.

That said, I think the way we have it now works fine. It is a bit cleaner to have properties as part of the interface (docstrings are nice, too), and it also makes it clear that plugin authors should implement the necessary methods to return the values needed.